File: String/TypeScript/_0389_FindtheDifference.ts
function findTheDifference(s: string, t: string): string {
  let res = "";
  let ls: string[] = s.split("").sort();
  let lt: string[] = t.split("").sort();

  for (let i = 0; i < lt.length; i++) {
    if (ls[i] != lt[i]) {
      res = lt[i];
      break;
    }
  }

  return res;
}

console.log(findTheDifference("abcd", "abcde"));
console.log(findTheDifference("", "y"));

File: String/Python/_0389_FindtheDifference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        ls = list(s)
        lt = list(t)

        ls.sort()
        lt.sort()

        for i in range(0, len(lt)):
            try:
                if ls[i] != lt[i]:
                    return lt[i]

            except IndexError:
                return lt[i]


ins = Solution()

print(ins.findTheDifference("abcd", "abcde"))
print(ins.findTheDifference("", "y"))
